### ISSUE TITLE
Smart Contract to Wallet

### ISSUE THREAD

**Issue by Razorholt** [kornrunner/php-ethereum-offline-raw-tx#7]
Can I use that script to send a specific ERC20 token from a smart contract to a wallet? Thanks!

**Comment by kornrunner** [comment]
Perhaps you're looking for #4 ?

**Comment by Razorholt** [comment]
> Perhaps you're looking for #4 ?

Thanks @kornrunner for your quick reply! Which package should I use to sign the transaction then? Do you have one?

**Comment by Razorholt** [comment]
Well, that must be very complicated or impossible to do because there are nothing out there explaining how to send a token from a contract to a wallet.

**Comment by kornrunner** [comment]
Yes, you can use this library for ERC20 transfers. The key is constructing the data field yourself.

  For an ERC20 `transfer(address,uint256)` call:
  - `to` as the token contract address
  - `value` is `0`
  -  `data` to the ABI-encoded function call

```php
  $data = '0xa9059cbb'  // transfer function selector
      . str_pad(substr($recipientAddress, 2), 64, '0', STR_PAD_LEFT)  // recipient (32 bytes)
      . str_pad(dechex($amount), 64, '0', STR_PAD_LEFT);  // amount (32 bytes)
```
This library handles signing - for ABI encoding you'd typically use a separate library like web3.php.
